service/smithyprototype/lexruntimeservice/deserialize.go

```diff
@@ -0,0 +1,1444 @@
+		return fmt.Errorf("unsupported deserialization of nil %T", v)
+	}
+
+	if v.ContentType != nil {
+		val := response.Header.Get("Content-Type")
+		v.ContentType = &val
+	}
+
+	if len(v.DialogState) > 0 {
+		val := response.Header.Get("x-amz-lex-dialog-state")
+		v.DialogState = types.DialogState(val)
+	}
+
+	if v.IntentName != nil {
+		val := response.Header.Get("x-amz-lex-intent-name")
+		v.IntentName = &val
+	}
+
+	if v.Message != nil {
+		val := response.Header.Get("x-amz-lex-message")
+		v.Message = &val
+	}
+
+	if len(v.MessageFormat) > 0 {
+		val := response.Header.Get("x-amz-lex-message-format")
+		v.MessageFormat = types.MessageFormatType(val)
+	}
+
+	if v.SessionAttributes != nil {
+		val := response.Header.Get("x-amz-lex-session-attributes")
+		b, err := base64.StdEncoding.DecodeString(val)
+		if err != nil {
+			return err
+		}
+		m := aws.JSONValue{}
+		err = json.Unmarshal(b, &m)
+		if err != nil {
+			return err
+		}
+		v.SessionAttributes = m
+	}
+
+	if v.SessionId != nil {
+		val := response.Header.Get("x-amz-lex-session-id")
+		v.SessionId = &val
+	}
+
+	if v.SlotToElicit != nil {
+		val := response.Header.Get("x-amz-lex-slot-to-elicit")
+		v.SlotToElicit = &val
+	}
+
+	if v.Slots != nil {
+		val := response.Header.Get("x-amz-lex-slots")
+		b, err := base64.StdEncoding.DecodeString(val)
+		if err != nil {
+			return err
+		}
+		m := aws.JSONValue{}
+		err = json.Unmarshal(b, &m)
+		if err != nil {
+			return err
+		}
+		v.Slots = m
+	}
+
+	return nil
+}
+
+func awsRestJson1_deserializeJsonPutSessionOutput(v *PutSessionOutput, body io.ReadCloser) error {
+	if v == nil {
+		return fmt.Errorf("unsupported deserialization of nil %T", v)
+	}
+
+	v.AudioStream = body
+	return nil
+}
+
+type awsRestJson1_deserializeOpPostText struct{}
+
+// ID is the middleware identifier
+func (p awsRestJson1_deserializeOpPostText) ID() string {
+	return "awsRestJson1_deserializeOpPostText"
+}
+
+func (p awsRestJson1_deserializeOpPostText) HandleDeserialize(ctx context.Context, in middleware.DeserializeInput, next middleware.DeserializeHandler) (
+	out middleware.DeserializeOutput, metadata middleware.Metadata, err error,
+) {
+	out, metadata, err = next.HandleDeserialize(ctx, in)
+	if err != nil {
+		return out, metadata, err
+	}
+	response, ok := out.RawResponse.(*smithyHTTP.Response)
+	if !ok {
+		return out, metadata, &aws.DeserializationError{
+			Err: fmt.Errorf("unknown transport type %T", out.RawResponse),
+		}
+	}
+
+	output, ok := out.Result.(*PostTextOutput)
+	if !ok {
+		return out, metadata, &aws.DeserializationError{
+			Err: fmt.Errorf("unknown output result type %T", out.Result),
+		}
+	}
+
+	buff := make([]byte, 1024)
+	ringBuffer := sdkio.NewRingBuffer(buff)
+
+	// wrap a TeeReader to read from response body & write on snapshot
+	body := io.TeeReader(response.Body, ringBuffer)
+	defer response.Body.Close()
+	decoder := json.NewDecoder(body)
+
+	if err := awsRestJson1_deserializeJsonPostTextOutput(output, decoder); err != nil {
```

Comment:
probably also want to configure the `json.Decoder` to decode all numbers as `json.Number` type. so we don't have to worry about it decoding them as float vs int. The generated SDK's deserialize logic can take care of converting the `json.Number` to the modeled type.

https://pkg.go.dev/encoding/json?tab=doc#Decoder.UseNumber

Reply:
Will update to use Json.Number instead; however that is an interface{} instead of float64 and will need type casting everytime, isn't that more expensive than type casting if int64?

Reply:
Will update to use json.Number